File: DiagonalPrincipal.java
import java.text.DecimalFormat;
import java.util.Scanner;

public class DiagonalPrincipal {
    
    public static void main(String[] args) {
        
        Scanner scanner = new Scanner(System.in);
        DecimalFormat df = new DecimalFormat("0.0");

        String operacao = scanner.next();
        int N = scanner.nextInt();
        
        int[][] matriz = new int[N][N];
        double soma = 0;
        double media;
        int num = 0;
        int cont = 0;

        for(int i=0; i<N; i++){
            for(int j=0; j<N; j++){
                
                matriz[i][j] = scanner.nextInt();   
            }
        }

        if("S".equals(operacao)){
            
            for(int i=0; i<N; i++){
                for(int j=0; j<N; j++){
                    if(i == j){
                        soma =  soma + matriz[i][j];
                    }
                }
            }
            System.out.printf("%.1f\n", soma);
        }
        else if("M".equals(operacao)){
            
            for(int i=0; i<N; i++){
                for(int j=0; j<N; j++){
                    if(i == j){
                        num = num + matriz[i][j];
                        cont++;
                    }
                }
            }
            media = (double)num/cont;
            System.out.printf("%.1f\n", media);
        }

    }

}

File: DiagonalPrincipal.cpp
#include <iostream>
#include <iomanip>
#include <vector>
using namespace std;

int main() {
    char operacao;
    int N;

    cin >> operacao;
    cin >> N;

    vector<vector<int>> matriz(N, vector<int>(N));

    // Lê a matriz
    for (int i = 0; i < N; i++) {
        for (int j = 0; j < N; j++) {
            cin >> matriz[i][j];
        }
    }

    double soma = 0.0;
    int cont = 0;

    // Processa a diagonal principal
    for (int i = 0; i < N; i++) {
        soma += matriz[i][i];
        cont++;
    }

    cout << fixed << setprecision(1);

    if (operacao == 'S') {
        cout << soma << endl;
    } else if (operacao == 'M') {
        cout << (soma / cont) << endl;
    }

    return 0;
}
